keras/layers/recurrent.py

```diff
@@ -152,6 +152,13 @@ def call(self, inputs, states, constants=None, training=None, **kwargs):
   def build(self, input_shape):
     if isinstance(input_shape, list):
       input_shape = input_shape[0]
+
+    def get_batch_input_shape(batch_size):
```

Comment:
This factoring seems confusing. Consider using `get_batch_input_shape(batch_size, dim)` and calling it using `partial`

Reply:
ok, I have fixed it.